projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.cs

```diff
@@ -60,35 +60,29 @@ private Connection InnerConnection
             }
         }
 
-        internal AutorecoveringConnection(ConnectionConfig config, IEndpointResolver endpoints)
+        internal AutorecoveringConnection(ConnectionConfig config, IEndpointResolver endpoints, Connection innerConnection)
         {
             _config = config;
             _endpoints = endpoints;
-        }
-
-        internal async ValueTask<IConnection> OpenAsync(CancellationToken cancellationToken)
-        {
-            IFrameHandler fh = await _endpoints.SelectOneAsync(_config.FrameHandlerFactoryAsync, cancellationToken)
-                .ConfigureAwait(false);
-            CreateInnerConnection(fh);
-            await _innerConnection.OpenAsync(cancellationToken)
-                .ConfigureAwait(false);
-            return this;
-        }
-
-        private void CreateInnerConnection(IFrameHandler frameHandler)
-        {
-            _innerConnection = new Connection(_config, frameHandler);
-
-            void onException(Exception exception, string context) =>
-                _innerConnection.OnCallbackException(CallbackExceptionEventArgs.Build(exception, context));
+            _innerConnection = innerConnection;
 
+            ConnectionShutdown += HandleConnectionShutdown;
             _recoverySucceededWrapper = new EventingWrapper<EventArgs>("OnConnectionRecovery", onException);
             _connectionRecoveryErrorWrapper = new EventingWrapper<ConnectionRecoveryErrorEventArgs>("OnConnectionRecoveryError", onException);
             _consumerTagChangeAfterRecoveryWrapper = new EventingWrapper<ConsumerTagChangedAfterRecoveryEventArgs>("OnConsumerRecovery", onException);
             _queueNameChangedAfterRecoveryWrapper = new EventingWrapper<QueueNameChangedAfterRecoveryEventArgs>("OnQueueRecovery", onException);
 
-            ConnectionShutdown += HandleConnectionShutdown;
+            void onException(Exception exception, string context) =>
+                _innerConnection.OnCallbackException(CallbackExceptionEventArgs.Build(exception, context));
+        }
+
+        internal static async ValueTask<AutorecoveringConnection> CreateAsync(ConnectionConfig config, IEndpointResolver endpoints, CancellationToken cancellationToken)
```

Comment:
By adjusting the constructor and creating it in this method, the nullablity can be avoided for the Connection.